templates/contribute/commit.md

```diff
@@ -41,7 +41,7 @@ For instance, it could be
 `<subject>` has the following constraints:
 
 - use imperative, present tense: "change" not "changed" nor "changes"
-- do not capitalize the first letter
+- capitalize the first letter
```

Comment:
I don't think this is a good idea. 

> these were about half/half upper and lower case. 

Looking at PR titles from the first half of 2024, the majority were lower case, with the causes of exceptions being mostly new contributors.

> The typography of English sentences

PR titles are not English sentences, for instance they do not allow a `.` at the end. Similarly the grammatical structure of an English sentence is also not relevant in the context of a PR title.

> The requirement for the PR description

PR descriptions have a very different purpose from PR titles, so this is not a fair comparison.

> The commit messages auto-generated by git when doing a merge ("Merged remote-tracking branch ...")

Similarly to the previous, commit messages are not prefixed by a type, and thus must follow a different convention. This is intentional, and not a reason to change either. Note that the commit messages auto-generated by bots on PRs as well as when making PRs follow the lowercase convention.

> The statu quo on other leanprover-community repos, eg this one.

Other repos do not have a style guide, whereas the status quo for the mathlib repo and the Lean repo is to follow the lowercase convention.

Reply:
In fact, looking at the roughly 14 thousand commits to mathlib4, less than 2 thousand of these were upper-cased. Thus this particular change is a pretty significant deviation from what is done in practice.